crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1950,6 +1956,7 @@ impl<'db> KnownInstanceType<'db> {
             Self::Union => KnownClass::SpecialForm,
             Self::NoReturn => KnownClass::SpecialForm,
             Self::Never => KnownClass::SpecialForm,
+            Self::Any => KnownClass::SpecialForm,
```

Comment:
When we adjust this PR to use unmodified typeshed, this will instead return `KnownClass::Object`